app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,501 @@
+
+    const selectActiveChannels = useCallback(() => {
+        switch (typeOfChannels) {
+            case ARCHIVED:
+                return archivedChannels;
+            case SHARED:
+                return sharedChannels;
+            default:
+                return channels;
+        }
+    }, [archivedChannels, channels, sharedChannels, typeOfChannels]);
+
+    const stopSearch = useCallback(() => {
+        setVisibleChannels(selectActiveChannels());
+        setTerm('');
+    }, [selectActiveChannels]);
+
+    const searchChannels = useCallback((text: string) => {
+        if (text) {
+            const active = selectActiveChannels();
+            const filtered = filterChannelsByTerm(active, text);
+            setTerm(text);
+            if (
+                filtered.length !== visibleChannels.length ||
+                filtered.reduce((shouldUpdate, c, i) => shouldUpdate || c.id !== visibleChannels[i].id, false)
+            ) {
+                setVisibleChannels(filtered);
+            }
+        } else {
+            stopSearch();
+        }
+    }, [selectActiveChannels, visibleChannels]);
+
+    const onPressDropdownElement = useCallback((channelType: string) => {
+        setTypeOfChannels(channelType);
+    }, []);
+
+    const renderItem = useCallback(({item}: {item: Channel}) => {
+        return (
+            <ChannelListRow
+                channel={item}
+                testID='more_channels.custom_list.channel_item'
+                onPress={onSelectChannel}
+            />
+        );
+
+    // All dependencies derived from onSelectChannel
+    }, [theme, canCreateChannels, currentUserId, currentTeamId]);
+
+    const renderLoading = useCallback(() => {
+        return (
+            <Loading
+                containerStyle={style.loadingContainer}
+                style={style.loading}
+                color={theme.buttonBg}
+            />
+        );
+
+    //Style is covered by the team
+    }, [theme]);
+
+    const renderNoResults = useCallback(() => {
+        if (term) {
+            return (
+                <View style={style.noResultContainer}>
+                    <FormattedText
+                        id='mobile.custom_list.no_results'
+                        defaultMessage='No Results'
+                        style={style.noResultText}
+                    />
+                </View>
+            );
+        }
+
+        return (
+            <View style={style.noResultContainer}>
+                <FormattedText
+                    id='more_channels.noMore'
+                    defaultMessage='No more channels to join'
+                    style={style.noResultText}
+                />
+            </View>
+        );
+    }, [style, term]);
+
+    const renderSeparator = useCallback(() => (
+        <View
+            style={{
+                height: 1,
+                backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+                width: '100%',
```

Comment:
Not sure if it is worth to move this to the styles. It should not create extra rerenders to pass the styles like this.